.travis.yml

```diff
@@ -31,60 +24,60 @@ matrix:
         - DEPS=lowest
     - php: 5.6
       env:
-        - DEPS=latest
+        - DEPS=locked
         - TEST_COVERAGE=true
-        - DEPLOY_DOCS="$(if [[ $TRAVIS_BRANCH == 'master' && $TRAVIS_PULL_REQUEST == 'false' ]]; then echo -n 'true' ; else echo -n 'false' ; fi)"
-        - PATH="$HOME/.local/bin:$PATH"
+    - php: 5.6
+      env:
+        - DEPS=latest
     - php: 7
       env:
         - DEPS=lowest
     - php: 7
       env:
-        - DEPS=latest
+        - DEPS=locked
         - CS_CHECK=true
+    - php: 7
+      env:
+        - DEPS=latest
     - php: 7.1
       env:
         - DEPS=lowest
     - php: 7.1
       env:
+        - DEPS=locked
+    - php: 7.1
+      env:
         - DEPS=latest
-    - php: hhvm
+    - php: 7.2
       env:
         - DEPS=lowest
-    - php: hhvm
+    - php: 7.2
+      env:
+        - DEPS=locked
+    - php: 7.2
       env:
         - DEPS=latest
   allow_failures:
-    - php: hhvm
+    - php: 7.2
 
 before_install:
-  - if [[ $TRAVIS_PHP_VERSION != "hhvm" && $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini ; fi
+  - if [[ $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
   - travis_retry composer self-update
 
 install:
-  - travis_retry composer install $COMPOSER_ARGS
+  - travis_retry composer install $COMPOSER_ARGS --ignore-platform-reqs
```

Comment:
You _are_ missing something; the lockfile is only relevant when you are running `composer install` _within a checkout of this package_; otherwise, the project _composing_ the package has its own lockfile that is used.

Essentially, the lockfile here is for use by _contributors_ and our _continuous integration_, and essentially ignored by _consumers_, as it's irrelevant at an application level.

@webimpress To make that last point clear, don't forget to add `composer.lock` to the `.gitattributes` file. :smile: